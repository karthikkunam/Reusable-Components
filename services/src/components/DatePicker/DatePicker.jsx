// This Component is a wrapper for https://ant.design/components/date-picker/
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';
import { DatePicker as AntdDatepicker } from 'antd';
import { printError } from '../../helpers/log';
import { isDisabledDate, getMapping, isSameArray } from '../../helpers/datePicker';

/* The below css files are needed to be imported saperately from antd for DatePicker because jest is unable to read css.js file
in antd Datepicker component for datePicker styles */
import 'antd/es/date-picker/style/index.css';
import 'antd/es/button/style/index.css';
import 'antd/es/tag/style/index.css';
import 'antd/es/style/index.css';
/* END of antd Styles */
import './datepicker.scss';

const defaultDayFormat = 'YYYY-MM-DD';
export default class DatePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: undefined,
      disableDatesMapping: {},
    };
  }

  updateSelectedDate = date => {
    this.setState({ value: date });
  };

  componentDidMount() {
    const { value, disableDatesList } = this.props;
    value && this.updateSelectedDate(value);
    // create disableDatesMapping if disableDatesList is received
    disableDatesList &&
      Array.isArray(disableDatesList) &&
      disableDatesList.length > 0 &&
      this.createDisableDatesMapping(disableDatesList);
  }

  componentDidUpdate(prevProps) {
    const { value: prevValue } = prevProps;
    const { value: currValue, disableDatesList } = this.props;
    const { disableDatesMapping } = this.state;

    const isDateChanged = !moment(prevValue).isSame(currValue);
    isDateChanged && this.updateSelectedDate(currValue);

    // Update disableDatesMapping if disableDatesList is updated
    if (
      disableDatesList &&
      Array.isArray(disableDatesList) &&
      !isSameArray({ mapping: disableDatesMapping, data: disableDatesList })
    ) {
      this.createDisableDatesMapping(disableDatesList);
    }
  }

  createDisableDatesMapping = data => {
    const mapping = getMapping(data, defaultDayFormat);
    this.setState({ disableDatesMapping: mapping });
  };

  handleChange = value => {
    const { onChange } = this.props;
    this.setState({ value });
    onChange && onChange(value);
  };

  /*
    The below function disables the selected dates.It recieves currentDate( which is everyday in a month ) and returns a bool.
    true  = disable currenDate
    false = enable currentDate
  */
  disabledDate = currentDate => {
    const { range = {}, disabledDate: disabledDateFn } = this.props;
    const { disableDatesMapping = {} } = this.state;

    // First check if disableDate function if received from props
    if (disabledDateFn) return disabledDateFn(currentDate);
    const args = {
      range,
      disableDatesMapping,
      currentDate,
    };
    return isDisabledDate(args);
  };

  getProps = () => {
    /* 
      className, dropdownClassName, popupStyle, bordered, suffixIcon, style, monthCellContentRender
      props are disabled as 7Boss Styles are applied 
    */
    /* eslint-disable */
    const {
      // props not allowed
      bordered,
      className,
      dropdownClassName,
      popupStyle,
      suffixIcon,
      style,
      disabledDate,
      range,
      disableDatesList,
      monthCellContentRender,
      // end of props not allowed
      ...rest
    } = this.props;
    /* eslint-enable */
    let { value } = this.state;
    const props = {
      onChange: this.handleChange,
      disabledDate: this.disabledDate,
      ...rest,
    };

    let { defaultValue } = this.props;

    // validate defaultValue and value
    if (defaultValue && !moment.isMoment(defaultValue)) {
      // print Error
      printError({ text: "Error In DatePicker: 'value' Passed is not a moment", defaultValue });
      defaultValue = moment();
    }
    if (value && !moment.isMoment(value)) {
      value = moment();
      // print Error
      printError({ text: "Error In DatePicker: 'value' Passed is not a moment", value });
    }

    props.value = value;
    props.defaultValue = defaultValue;

    return props;
  };

  render() {
    const allProps = this.getProps();
    return (
      <div className="ant-picker-container">
        <label className="ant-picker-label">Start Date</label>
        <AntdDatepicker {...allProps} />
      </div>
    );
  }
}

DatePicker.defaultProps = {
  showToday: false,
  format: 'MM/DD/YYYY',
};

DatePicker.propTypes = {
  allowClear: PropTypes.bool,
  autoFocus: PropTypes.bool,
  dateRender: PropTypes.func,
  disabled: PropTypes.bool,
  disabledTime: PropTypes.func,
  disabledDate: PropTypes.func,
  disableDatesList: PropTypes.array,
  defaultValue: PropTypes.instanceOf(moment),
  format: PropTypes.string,
  mode: PropTypes.string,
  open: PropTypes.bool,
  onChange: PropTypes.func,
  onOpenChange: PropTypes.func,
  onPanelChange: PropTypes.func,
  onOk: PropTypes.func,
  picker: PropTypes.string,
  placeholder: PropTypes.string,
  range: PropTypes.object,
  renderExtraFooter: PropTypes.func,
  size: PropTypes.string,
  showToday: PropTypes.bool,
  showTime: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
  value: PropTypes.instanceOf(moment),
};
