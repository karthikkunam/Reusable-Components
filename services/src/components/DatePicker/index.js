import DatePicker from './DatePicker';
import RangePicker from './RangePicker';

export { DatePicker, RangePicker };
