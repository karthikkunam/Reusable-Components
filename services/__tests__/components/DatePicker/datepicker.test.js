import React from 'react';
import moment from 'moment-timezone';
import { shallow, mount } from 'enzyme';
import DatePicker from '../../../src/components/DatePicker/DatePicker';

let wrapper = null;
const date = '2020-03-09';

beforeEach(() => {
  wrapper = mount(<DatePicker />);
});

describe('DatePicker Tests', () => {
  it('renders correctly', () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('renders with default value', () => {
    wrapper.setProps({ defaultValue: moment(date) });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('disabled', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('allowClear', () => {
    wrapper.setProps({ allowClear: true });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('autoFocus', () => {
    wrapper.setProps({ autoFocus: true });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('dateRender', () => {
    const dateRender = currentDate => {
      let style = {};
      if (currentDate.date() === 1) {
        style.border = '1px solid #107f62';
        style.borderRadius = '50%';
      }
      return (
        <div className="ant-picker-cell-inner" style={style}>
          {currentDate.date()}
        </div>
      );
    };
    wrapper.setProps({ dateRender });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('showTime', () => {
    wrapper.setProps({ showTime: true });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('format', () => {
    wrapper.setProps({ format: 'MM/DD/YY' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('mode', () => {
    wrapper.setProps({ mode: 'year' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('picker', () => {
    wrapper.setProps({ picker: 'year' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('onOpenChange', () => {
    wrapper.setProps({ onOpenChange: status => console.log('onOpenChange', status) });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('onPanelChange', () => {
    wrapper.setProps({ onPanelChange: (value, mode) => console.log('onPanelChange', value, mode) });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('onOk', () => {
    wrapper.setProps({ showTime: true, onOk: () => console.log('onOk') });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('placeholder', () => {
    wrapper.setProps({ placeholder: 'Pick Date' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('renderExtraFooter', () => {
    const renderExtraFooter = () => <div />;
    wrapper.setProps({ renderExtraFooter });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('size', () => {
    wrapper.setProps({ size: 'large' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('showToday', () => {
    wrapper.setProps({ showToday: true });
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('Check if Value in Component state is Updated', () => {
    const shallowWrapper = shallow(<DatePicker />);
    const instance = shallowWrapper.instance();
    expect(shallowWrapper.state('value')).toBe(undefined);
    const newDate = moment();
    instance.updateSelectedDate(newDate);
    expect(moment(shallowWrapper.state('value')).isSame(newDate, 'day')).toBeTruthy();
  });
});
