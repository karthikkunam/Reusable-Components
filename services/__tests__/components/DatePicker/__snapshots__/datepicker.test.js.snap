// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Tests allowClear 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests autoFocus 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        autofocus=""
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests dateRender 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests disabled 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker ant-picker-disabled"
  >
    <div
      class="ant-picker-input"
    >
      <input
        disabled=""
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests format 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests mode 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests onOk 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests onOpenChange 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests onPanelChange 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests picker 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select year"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests placeholder 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Pick Date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests renderExtraFooter 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests renders correctly 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests renders with default value 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests showTime 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests showToday 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Tests size 1`] = `
<div
  class="ant-picker-container"
>
  <label
    class="ant-picker-label"
  >
    Start Date
  </label>
  <div
    class="ant-picker ant-picker-large"
  >
    <div
      class="ant-picker-input"
    >
      <input
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;
